modules/services/workspace-service/src/main/java/org/ballerinalang/composer/service/workspace/rest/datamodel/BLangFileRestService.java

```diff
@@ -167,39 +168,43 @@ private static String parseJsonDataModel(BFile bFile) throws IOException, Invoca
         String filePath = bFile.getFilePath();
         String fileName = bFile.getFileName();
         String content = bFile.getContent();
-
+    
         org.wso2.ballerinalang.compiler.tree.BLangPackage model;
         List<Diagnostic> diagnostics;
-
+    
         // Sometimes we are getting Ballerina content without a file in the file-system.
         if (!Files.exists(Paths.get(filePath, fileName))) {
             model = WorkspaceUtils.getBallerinaFileForContent(fileName, content, CompilerPhase.CODE_ANALYZE)
                     .getBLangPackage();
             diagnostics = WorkspaceUtils.getBallerinaFileForContent(fileName, content, CompilerPhase.CODE_ANALYZE)
                     .getDiagnostics();
-
-        }else{
+        
+        } else {
             model = WorkspaceUtils.getBallerinaFile(filePath, fileName).getBLangPackage();
             diagnostics = WorkspaceUtils.getBallerinaFile(fileName, content).getDiagnostics();
         }
-
+    
         final Map<String, ModelPackage> modelPackage = new HashMap<>();
         WorkspaceUtils.loadPackageMap("Current Package", model, modelPackage);
-
+    
         Gson gson = new Gson();
         JsonElement diagnosticsJson = gson.toJsonTree(diagnostics);
-        JsonElement packageInfo = gson.toJsonTree(modelPackage.values().stream().findFirst().get());
-
+    
         BLangCompilationUnit compilationUnit = model.getCompilationUnits().stream().
                 filter(compUnit -> fileName.equals(compUnit.getName())).findFirst().get();
         JsonElement modelElement = generateJSON(compilationUnit);
         JsonObject result = new JsonObject();
         result.add("model", modelElement);
         result.add("diagnostics", diagnosticsJson);
-        result.add("packageInfo", packageInfo);
-
+    
+        // Add 'packageInfo' only if there are any packages.
+        Optional<ModelPackage> packageInfoJson = modelPackage.values().stream().findFirst();
+        if (packageInfoJson.isPresent()) {
```

Comment:
@sumuditha-viraj , packageInfo will be passed only if it is available. I check the JS code, we have a null check there[1]. So I think it will be fine to send if packageInfo is available.

[1] - https://github.com/ballerinalang/composer/blob/semantic-refactor/modules/web/js/ballerina/views/ballerina-file-editor.jsx#L405

Reply:
+1